ccl/multiregionccl: TestMrSystemDatabaseUpgrade failed

ccl/multiregionccl.TestMrSystemDatabaseUpgrade [failed](https://mesolite.cluster.engflow.com/invocations/default/4e19db1d-184d-4b88-983f-5e1c81de2d94?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [c40ddee24c763738c6df9829d1a41d72a78aebec](https://github.com/cockroachdb/cockroach/commits/c40ddee24c763738c6df9829d1a41d72a78aebec):

Fatal error:

```
panic: markers.RedactableString("unable to insert duplicate lease for id=104") [recovered]
	panic: markers.RedactableString("unable to insert duplicate lease for id=104")

```
Stack: 

```
goroutine 53019 [running]:
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build.func1()
	pkg/sql/opt/optbuilder/builder.go:256 +0xfa
panic({0x6d43e00?, 0xc01146a6c0?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorSet).insert(0xc00c2792e0, 0xc00f1904c0)
	pkg/sql/catalog/lease/descriptor_set.go:47 +0x25f
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).insertDescriptorVersions(0xc004b0db88, {0x86da598, 0xc00e5b8be0}, 0x68?, {0xc010d6f600, 0x3, 0x0?})
	pkg/sql/catalog/lease/lease.go:975 +0x3bf
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).Acquire(0xc004b0db88, {0x86da598, 0xc00e5b8be0}, {0x68?, 0x0?}, 0x68)
	pkg/sql/catalog/lease/lease.go:1744 +0x17c
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*leasedDescriptors).getByID(0xc018cc6868, {0x86da598, 0xc00e5b8be0}, {0x8694278, 0xc00e378c60}, 0x68)
	pkg/sql/catalog/descs/leased_descriptors.go:214 +0x90
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*byIDLookupContext).lookupLeased(0xc01314eef0, 0x68)
	pkg/sql/catalog/descs/descriptor.go:373 +0xbd
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.getDescriptorsByID({0x86da598, 0xc00e5b8be0}, 0xc018cc6840, 0xc00e378c60, {{0x0, 0x0}, {0x1, 0x0, 0x0, 0x0}, ...}, ...)
	pkg/sql/catalog/descs/descriptor.go:175 +0xb9b
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.ByIDGetter.Desc({{0xc00e378c60, 0xc018cc6840}, {{0x0, 0x0}, {0x1, 0x0, 0x0, 0x0}, {0x1, 0x0, ...}}}, ...)
	pkg/sql/catalog/descs/getters.go:43 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.makeImmutableTypeLookupFunc.func1({0x86da598?, 0xc00e5b8be0?}, 0x204869e?, 0x0?)
	pkg/sql/catalog/descs/hydrate.go:210 +0x7c
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.makeImmutableTypeLookupFunc.makeTypeLookupFuncForHydration.func2({0x86da598, 0xc00e5b8be0}, 0x68)
	pkg/sql/catalog/descs/hydrate.go:307 +0xc2
github.com/cockroachdb/cockroach/pkg/sql/catalog/typedesc.TypeLookupFunc.GetTypeDescriptor(0xfe9dca?, {0x86da598?, 0xc00e5b8be0?}, 0xc84b890?)
	pkg/sql/catalog/typedesc/type_desc.go:751 +0x52
github.com/cockroachdb/cockroach/pkg/sql/catalog/hydrateddesccache.canUseCachedDescriptor({0x86da598, 0xc00e5b8be0}, 0x0?, {0x867d700, 0xc011b713a0})
	pkg/sql/catalog/hydrateddesccache/hydratedcache.go:375 +0xdb
github.com/cockroachdb/cockroach/pkg/sql/catalog/hydrateddesccache.(*Cache).GetHydratedDescriptor(0xc0054b4070, {0x86da598, 0xc00e5b8be0}, {0x7fb148b89b78, 0xc00f100008}, {0x867d700, 0xc011b713a0})
	pkg/sql/catalog/hydrateddesccache/hydratedcache.go:300 +0x216
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*Collection).hydrateDescriptors(0xc018cc6840, {0x86da598, 0xc00e5b8be0}, 0xc00e378c60, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, ...}, ...)
	pkg/sql/catalog/descs/hydrate.go:91 +0x58a
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.getDescriptorsByID({0x86da598, 0xc00e5b8be0}, 0xc018cc6840, 0xc00e378c60, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, ...}, ...)
	pkg/sql/catalog/descs/descriptor.go:226 +0xad3
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.ByIDGetter.Desc({{0xc00e378c60, 0xc018cc6840}, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, {0x1, 0x1, ...}}}, ...)
	pkg/sql/catalog/descs/getters.go:43 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.ByIDGetter.Table({{0xc00e378c60, 0xc018cc6840}, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, {0x1, 0x1, ...}}}, ...)
	pkg/sql/catalog/descs/getters.go:85 +0x9a
github.com/cockroachdb/cockroach/pkg/sql.(*planner).LookupTableByID(0x0?, {0x86da598?, 0xc00e5b8be0?}, 0x0?)
	pkg/sql/planner.go:763 +0x75
github.com/cockroachdb/cockroach/pkg/sql.(*optCatalog).ResolveDataSourceByID(0xc0103ae320, {0x86da598, 0xc00e5b8be0}, {0x0?, 0x0?, 0x0?, 0x0?}, 0x9)
	pkg/sql/opt_catalog.go:362 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).resolveDataSourceRef(0xc018cc6b00, 0xc0103ae2d0, 0x5)
	pkg/sql/opt/optbuilder/util.go:696 +0xa8
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildCreateStatistics(0xc018cc6b00, 0xc00d57b200, 0x60?)
	pkg/sql/opt/optbuilder/misc_statements.go:170 +0x1d6
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc018cc6b00, {0x8702a40, 0xc00d57b200}, {0x0, 0x0, 0x0}, 0xc019f9e008)
	pkg/sql/opt/optbuilder/builder.go:469 +0x338
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmtAtRootWithScope(0xc018cc6b00, {0x8702a40, 0xc00d57b200}, {0x0, 0x0, 0x0}, 0xc019f9e008)
	pkg/sql/opt/optbuilder/builder.go:331 +0x1d0
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmtAtRoot(0xc018cc6b00?, {0x8702a40?, 0xc00d57b200?}, {0x0?, 0x0?, 0x0?})
	pkg/sql/opt/optbuilder/builder.go:307 +0x118
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build(0xc018cc6b00)
	pkg/sql/opt/optbuilder/builder.go:286 +0x2c8
github.com/cockroachdb/cockroach/pkg/sql.(*optPlanningCtx).buildExecMemo(0xc00f161608, {0x86da598, 0xc00e5b8be0})
	pkg/sql/plan_opt.go:855 +0x253
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc00f160728, {0x86da598?, 0xc00e5b8b40?})
	pkg/sql/plan_opt.go:261 +0x189
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc00f160008, {0x86da598, 0xc00e5b8b40}, 0xc00f160728)
	pkg/sql/conn_executor_exec.go:3209 +0xc6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00f160008, {0x86da598, 0xc00e5b8b40}, 0xc00f160728, {0x8769240, 0xc0103ae460})
	pkg/sql/conn_executor_exec.go:2786 +0x9fd
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00f160008, {0x86d78a0?, 0xc0103ae410?}, {{0x8702a40, 0xc00d57b200}, {0x0, 0x0, 0x0}, {0xc0089aac60, 0x58}, ...}, ...)
	pkg/sql/conn_executor_exec.go:1071 +0x386c
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func2({0x86d78a0?, 0xc0103ae410?})
	pkg/sql/conn_executor_exec.go:173 +0xb0
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0x86d78a0?, {0x86d78a0?, 0xc0103ae410?}, {0x8702a40?, 0xc00d57b200?}, 0x7fb13c989aa0?, 0x500?)
	pkg/sql/conn_executor_exec.go:4400 +0x2ec
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00f160008, {0x86d78a0, 0xc0103ae410}, {{0x8702a40, 0xc00d57b200}, {0x0, 0x0, 0x0}, {0xc0089aac60, 0x58}, ...}, ...)
	pkg/sql/conn_executor_exec.go:172 +0x8e5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1({{{0x8702a40, 0xc00d57b200}, {0x0, 0x0, 0x0}, {0xc0089aac60, 0x58}, 0x0, 0x0}, 0x112febc947, ...}, ...)
	pkg/sql/conn_executor.go:2371 +0x3dd
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00f160008)
	pkg/sql/conn_executor.go:2376 +0x888
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00f160008, {0x86da598, 0xc00e5b8aa0}, 0x17d6746?, 0x17cfd14?, 0x0?)
	pkg/sql/conn_executor.go:2284 +0x208
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx.func2()
	pkg/sql/internal.go:258 +0x168
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx in goroutine 23390
	pkg/sql/internal.go:250 +0x2c5
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestMrSystemDatabaseUpgrade
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestMrSystemDatabaseUpgrade1909127069
    test_log_scope.go:82: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMrSystemDatabaseUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54380

ccl/multiregionccl.TestMrSystemDatabaseUpgrade [failed](https://mesolite.cluster.engflow.com/invocations/default/b18573f8-07a5-4360-b8bb-af708884d82b?testReportRun=7&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [223617d5f65835071f1f3395bc7b487e6dc029c9](https://github.com/cockroachdb/cockroach/commits/223617d5f65835071f1f3395bc7b487e6dc029c9):

Fatal error:

```
panic: markers.RedactableString("unable to insert duplicate lease for id=104") [recovered]
	panic: markers.RedactableString("unable to insert duplicate lease for id=104")

```
Stack: 

```
goroutine 91314 [running]:
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build.func1()
	pkg/sql/opt/optbuilder/builder.go:256 +0xfa
panic({0x6d8acc0?, 0xc01ae65960?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorSet).insert(0xc00e47cd90, 0xc01b106a40)
	pkg/sql/catalog/lease/descriptor_set.go:47 +0x25f
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).insertDescriptorVersions(0xc002822008, {0x87301f8, 0xc0161ab7c0}, 0x68?, {0xc01affe600, 0x3, 0x0?})
	pkg/sql/catalog/lease/lease.go:981 +0x3bf
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).Acquire(0xc002822008, {0x87301f8, 0xc0161ab7c0}, {0x86ea230, 0xc01a7b5700}, 0x68)
	pkg/sql/catalog/lease/lease.go:1815 +0x18e
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*leasedDescriptors).getByID(0xc0160b70a8, {0x87301f8, 0xc0161ab7c0}, {0x8706110, 0xc01b06b810}, 0x68)
	pkg/sql/catalog/descs/leased_descriptors.go:237 +0x9b
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*byIDLookupContext).lookupLeased(0xc01b056f48, 0x68)
	pkg/sql/catalog/descs/descriptor.go:373 +0xbd
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.getDescriptorsByID({0x87301f8, 0xc0161ab7c0}, 0xc0160b7080, 0xc01b06b810, {{0x0, 0x0}, {0x1, 0x0, 0x0, 0x0}, ...}, ...)
	pkg/sql/catalog/descs/descriptor.go:175 +0xb9b
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.ByIDGetter.Desc({{0xc01b06b810, 0xc0160b7080}, {{0x0, 0x0}, {0x1, 0x0, 0x0, 0x0}, {0x1, 0x0, ...}}}, ...)
	pkg/sql/catalog/descs/getters.go:43 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.makeImmutableTypeLookupFunc.func1({0x87301f8?, 0xc0161ab7c0?}, 0x2059ebe?, 0x0?)
	pkg/sql/catalog/descs/hydrate.go:210 +0x7c
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.makeImmutableTypeLookupFunc.makeTypeLookupFuncForHydration.func2({0x87301f8, 0xc0161ab7c0}, 0x68)
	pkg/sql/catalog/descs/hydrate.go:307 +0xc2
github.com/cockroachdb/cockroach/pkg/sql/catalog/typedesc.TypeLookupFunc.GetTypeDescriptor(0xfea00a?, {0x87301f8?, 0xc0161ab7c0?}, 0x1dccd80?)
	pkg/sql/catalog/typedesc/type_desc.go:751 +0x52
github.com/cockroachdb/cockroach/pkg/sql/catalog/hydrateddesccache.canUseCachedDescriptor({0x87301f8, 0xc0161ab7c0}, 0x0?, {0x86d2d80, 0xc01b066680})
	pkg/sql/catalog/hydrateddesccache/hydratedcache.go:375 +0xdb
github.com/cockroachdb/cockroach/pkg/sql/catalog/hydrateddesccache.(*Cache).GetHydratedDescriptor(0xc008bdaee0, {0x87301f8, 0xc0161ab7c0}, {0x7f2725d1bca0, 0xc01aedd208}, {0x86d2d80, 0xc01b066680})
	pkg/sql/catalog/hydrateddesccache/hydratedcache.go:300 +0x216
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*Collection).hydrateDescriptors(0xc0160b7080, {0x87301f8, 0xc0161ab7c0}, 0xc01b06b810, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, ...}, ...)
	pkg/sql/catalog/descs/hydrate.go:91 +0x58a
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.getDescriptorsByID({0x87301f8, 0xc0161ab7c0}, 0xc0160b7080, 0xc01b06b810, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, ...}, ...)
	pkg/sql/catalog/descs/descriptor.go:226 +0xad3
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.ByIDGetter.Desc({{0xc01b06b810, 0xc0160b7080}, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, {0x1, 0x1, ...}}}, ...)
	pkg/sql/catalog/descs/getters.go:43 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.ByIDGetter.Table({{0xc01b06b810, 0xc0160b7080}, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, {0x1, 0x1, ...}}}, ...)
	pkg/sql/catalog/descs/getters.go:85 +0x9a
github.com/cockroachdb/cockroach/pkg/sql.(*planner).LookupTableByID(0x0?, {0x87301f8?, 0xc0161ab7c0?}, 0x0?)
	pkg/sql/planner.go:765 +0x75
github.com/cockroachdb/cockroach/pkg/sql.(*optCatalog).ResolveDataSourceByID(0xc01b05d130, {0x87301f8, 0xc0161ab7c0}, {0x0?, 0x0?, 0x0?, 0x0?}, 0x27)
	pkg/sql/opt_catalog.go:362 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).resolveDataSourceRef(0xc01b0886e0, 0xc01b05d0e0, 0x5)
	pkg/sql/opt/optbuilder/util.go:696 +0xa8
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildCreateStatistics(0xc01b0886e0, 0xc0161a77a0, 0x60?)
	pkg/sql/opt/optbuilder/misc_statements.go:170 +0x1d6
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc01b0886e0, {0x8759460, 0xc0161a77a0}, {0x0, 0x0, 0x0}, 0xc01b02c908)
	pkg/sql/opt/optbuilder/builder.go:469 +0x338
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmtAtRootWithScope(0xc01b0886e0, {0x8759460, 0xc0161a77a0}, {0x0, 0x0, 0x0}, 0xc01b02c908)
	pkg/sql/opt/optbuilder/builder.go:331 +0x1d0
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmtAtRoot(0xc01b0886e0?, {0x8759460?, 0xc0161a77a0?}, {0x0?, 0x0?, 0x0?})
	pkg/sql/opt/optbuilder/builder.go:307 +0x118
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build(0xc01b0886e0)
	pkg/sql/opt/optbuilder/builder.go:286 +0x2c8
github.com/cockroachdb/cockroach/pkg/sql.(*optPlanningCtx).buildExecMemo(0xc01b08f608, {0x87301f8, 0xc0161ab7c0})
	pkg/sql/plan_opt.go:855 +0x253
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc01b08e728, {0x87301f8?, 0xc0161ab720?})
	pkg/sql/plan_opt.go:261 +0x189
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc01b08e008, {0x87301f8, 0xc0161ab720}, 0xc01b08e728)
	pkg/sql/conn_executor_exec.go:3209 +0xc6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc01b08e008, {0x87301f8, 0xc0161ab720}, 0xc01b08e728, {0x87c20a0, 0xc01b05d270})
	pkg/sql/conn_executor_exec.go:2786 +0x9fd
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenStateWithPausablePortal(0xc01b08e008, {0x872d500, 0xc01b05d220}, {{0x8759460, 0xc0161a77a0}, {0x0, 0x0, 0x0}, {0xc015936c60, 0x59}, ...}, ...)
	pkg/sql/conn_executor_exec.go:2045 +0x487b
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1({0x872d500?, 0xc01b05d220?})
	pkg/sql/conn_executor_exec.go:166 +0xb0
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0xcd6a2e0?, {0x872d500?, 0xc01b05d220?}, {0x8759460?, 0xc0161a77a0?}, 0x7f2724fe0220?, 0x500?)
	pkg/sql/conn_executor_exec.go:4400 +0x2ec
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc01b08e008, {0x872d500, 0xc01b05d220}, {{0x8759460, 0xc0161a77a0}, {0x0, 0x0, 0x0}, {0xc015936c60, 0x59}, ...}, ...)
	pkg/sql/conn_executor_exec.go:165 +0x79c
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1({{{0x8759460, 0xc0161a77a0}, {0x0, 0x0, 0x0}, {0xc015936c60, 0x59}, 0x0, 0x0}, 0x11b5e10980, ...}, ...)
	pkg/sql/conn_executor.go:2372 +0x3dd
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc01b08e008)
	pkg/sql/conn_executor.go:2377 +0x888
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc01b08e008, {0x87301f8, 0xc0161ab680}, 0x17e0b86?, 0xc01ae832d0?, 0x0?)
	pkg/sql/conn_executor.go:2285 +0x208
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx.func2()
	pkg/sql/internal.go:258 +0x168
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx in goroutine 39309
	pkg/sql/internal.go:250 +0x2c5
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestMrSystemDatabaseUpgrade
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestMrSystemDatabaseUpgrade342112085
    test_log_scope.go:82: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMrSystemDatabaseUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/multiregionccl.TestMrSystemDatabaseUpgrade [failed](https://mesolite.cluster.engflow.com/invocations/default/13fa4421-9643-4de1-afa6-fe7e0de00649?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [eb654e47a0482f948aa19d611422a23984213a64](https://github.com/cockroachdb/cockroach/commits/eb654e47a0482f948aa19d611422a23984213a64):

Fatal error:

```
panic: markers.RedactableString("unable to insert duplicate lease for id=104") [recovered]
	panic: markers.RedactableString("unable to insert duplicate lease for id=104")

```
Stack: 

```
goroutine 118270 [running]:
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build.func1()
	pkg/sql/opt/optbuilder/builder.go:256 +0xfa
panic({0x6d8e280?, 0xc002678000?})
	GOROOT/src/runtime/panic.go:791 +0x132
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorSet).insert(0xc009ebd420, 0xc00b831540)
	pkg/sql/catalog/lease/descriptor_set.go:47 +0x25f
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).insertDescriptorVersions(0xc001eb62c8, {0x8733cf8, 0xc00c813860}, 0x68?, {0xc0091a0f80, 0x3, 0x0?})
	pkg/sql/catalog/lease/lease.go:981 +0x3bf
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).Acquire(0xc001eb62c8, {0x8733cf8, 0xc00c813860}, {0x86edd30, 0xc0125e0080}, 0x68)
	pkg/sql/catalog/lease/lease.go:1815 +0x18e
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*leasedDescriptors).getByID(0xc004abe4a8, {0x8733cf8, 0xc00c813860}, {0x8709c10, 0xc00a73f600}, 0x68)
	pkg/sql/catalog/descs/leased_descriptors.go:237 +0x9b
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*byIDLookupContext).lookupLeased(0xc00f670f40, 0x68)
	pkg/sql/catalog/descs/descriptor.go:373 +0xbd
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.getDescriptorsByID({0x8733cf8, 0xc00c813860}, 0xc004abe480, 0xc00a73f600, {{0x0, 0x0}, {0x1, 0x0, 0x0, 0x0}, ...}, ...)
	pkg/sql/catalog/descs/descriptor.go:175 +0xb9b
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.ByIDGetter.Desc({{0xc00a73f600, 0xc004abe480}, {{0x0, 0x0}, {0x1, 0x0, 0x0, 0x0}, {0x1, 0x0, ...}}}, ...)
	pkg/sql/catalog/descs/getters.go:43 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.makeImmutableTypeLookupFunc.func1({0x8733cf8?, 0xc00c813860?}, 0x20595de?, 0x0?)
	pkg/sql/catalog/descs/hydrate.go:210 +0x7c
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.makeImmutableTypeLookupFunc.makeTypeLookupFuncForHydration.func2({0x8733cf8, 0xc00c813860}, 0x68)
	pkg/sql/catalog/descs/hydrate.go:307 +0xc2
github.com/cockroachdb/cockroach/pkg/sql/catalog/typedesc.TypeLookupFunc.GetTypeDescriptor(0xfe9e4a?, {0x8733cf8?, 0xc00c813860?}, 0x21fecd80?)
	pkg/sql/catalog/typedesc/type_desc.go:751 +0x52
github.com/cockroachdb/cockroach/pkg/sql/catalog/hydrateddesccache.canUseCachedDescriptor({0x8733cf8, 0xc00c813860}, 0x0?, {0x86d6880, 0xc0132b4700})
	pkg/sql/catalog/hydrateddesccache/hydratedcache.go:375 +0xdb
github.com/cockroachdb/cockroach/pkg/sql/catalog/hydrateddesccache.(*Cache).GetHydratedDescriptor(0xc008cb2af0, {0x8733cf8, 0xc00c813860}, {0x7f0eed350a28, 0xc00880c608}, {0x86d6880, 0xc0132b4700})
	pkg/sql/catalog/hydrateddesccache/hydratedcache.go:300 +0x216
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.(*Collection).hydrateDescriptors(0xc004abe480, {0x8733cf8, 0xc00c813860}, 0xc00a73f600, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, ...}, ...)
	pkg/sql/catalog/descs/hydrate.go:91 +0x58a
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.getDescriptorsByID({0x8733cf8, 0xc00c813860}, 0xc004abe480, 0xc00a73f600, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, ...}, ...)
	pkg/sql/catalog/descs/descriptor.go:226 +0xad3
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.ByIDGetter.Desc({{0xc00a73f600, 0xc004abe480}, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, {0x1, 0x1, ...}}}, ...)
	pkg/sql/catalog/descs/getters.go:43 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/catalog/descs.ByIDGetter.Table({{0xc00a73f600, 0xc004abe480}, {{0x0, 0x0}, {0x0, 0x0, 0x0, 0x0}, {0x1, 0x1, ...}}}, ...)
	pkg/sql/catalog/descs/getters.go:85 +0x9a
github.com/cockroachdb/cockroach/pkg/sql.(*planner).LookupTableByID(0x0?, {0x8733cf8?, 0xc00c813860?}, 0x0?)
	pkg/sql/planner.go:765 +0x75
github.com/cockroachdb/cockroach/pkg/sql.(*optCatalog).ResolveDataSourceByID(0xc0100890e0, {0x8733cf8, 0xc00c813860}, {0x0?, 0x0?, 0x0?, 0x0?}, 0x9)
	pkg/sql/opt_catalog.go:362 +0xe5
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).resolveDataSourceRef(0xc00d875760, 0xc010089090, 0x5)
	pkg/sql/opt/optbuilder/util.go:696 +0xa8
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildCreateStatistics(0xc00d875760, 0xc0110fe780, 0x60?)
	pkg/sql/opt/optbuilder/misc_statements.go:170 +0x1d6
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00d875760, {0x875d080, 0xc0110fe780}, {0x0, 0x0, 0x0}, 0xc0175a4d88)
	pkg/sql/opt/optbuilder/builder.go:469 +0x338
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmtAtRootWithScope(0xc00d875760, {0x875d080, 0xc0110fe780}, {0x0, 0x0, 0x0}, 0xc0175a4d88)
	pkg/sql/opt/optbuilder/builder.go:331 +0x1d0
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmtAtRoot(0xc00d875760?, {0x875d080?, 0xc0110fe780?}, {0x0?, 0x0?, 0x0?})
	pkg/sql/opt/optbuilder/builder.go:307 +0x118
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build(0xc00d875760)
	pkg/sql/opt/optbuilder/builder.go:286 +0x2c8
github.com/cockroachdb/cockroach/pkg/sql.(*optPlanningCtx).buildExecMemo(0xc0198a25e0, {0x8733cf8, 0xc00c813860})
	pkg/sql/plan_opt.go:855 +0x253
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc0198a16f8, {0x8733cf8?, 0xc00c8137c0?})
	pkg/sql/plan_opt.go:261 +0x189
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc0198a0f08, {0x8733cf8, 0xc00c8137c0}, 0xc0198a16f8)
	pkg/sql/conn_executor_exec.go:3195 +0xc6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc0198a0f08, {0x8733cf8, 0xc00c8137c0}, 0xc0198a16f8, {0x87c5ba0, 0xc010089220})
	pkg/sql/conn_executor_exec.go:2772 +0x9fd
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenStateWithPausablePortal(0xc0198a0f08, {0x8731000, 0xc0100891d0}, {{0x875d080, 0xc0110fe780}, {0x0, 0x0, 0x0}, {0xc013c1af60, 0x58}, ...}, ...)
	pkg/sql/conn_executor_exec.go:2031 +0x481b
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1({0x8731000?, 0xc0100891d0?})
	pkg/sql/conn_executor_exec.go:166 +0xb0
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0xcd702e0?, {0x8731000?, 0xc0100891d0?}, {0x875d080?, 0xc0110fe780?}, 0x7f0ec401a508?, 0x500?)
	pkg/sql/conn_executor_exec.go:4386 +0x2ec
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc0198a0f08, {0x8731000, 0xc0100891d0}, {{0x875d080, 0xc0110fe780}, {0x0, 0x0, 0x0}, {0xc013c1af60, 0x58}, ...}, ...)
	pkg/sql/conn_executor_exec.go:165 +0x79c
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1({{{0x875d080, 0xc0110fe780}, {0x0, 0x0, 0x0}, {0xc013c1af60, 0x58}, 0x0, 0x0}, 0x1a59df7a06, ...}, ...)
	pkg/sql/conn_executor.go:2375 +0x3dd
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc0198a0f08)
	pkg/sql/conn_executor.go:2380 +0x888
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc0198a0f08, {0x8733cf8, 0xc00c813720}, 0x17e0d46?, 0x17da314?, 0x0?)
	pkg/sql/conn_executor.go:2288 +0x208
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx.func2()
	pkg/sql/internal.go:258 +0x168
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx in goroutine 37197
	pkg/sql/internal.go:250 +0x2c5
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestMrSystemDatabaseUpgrade
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestMrSystemDatabaseUpgrade4243558933
    test_log_scope.go:82: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMrSystemDatabaseUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/upgrade/upgrades/first_upgrade.go
+++ b/pkg/upgrade/upgrades/first_upgrade.go
@@ -127,7 +127,11 @@ func upgradeDescriptors(
 				}
 				b := txn.KV().NewBatch()
 				for _, mut := range muts {
-					if !mut.GetPostDeserializationChanges().HasChanges() {
+					// Both newly created and altered descriptors never write the modification time
+					// to storage. This post-deserialization change is always expected now.
+					changes := mut.GetPostDeserializationChanges()
+					hasChanges := changes.Len() > 1 || !changes.Contains(catalog.SetModTimeToMVCCTimestamp)
+					if !hasChanges {
 						// In the upgrade to 25.4, we do a one-time rewrite of all
 						// descriptors in order to upgrade them to use the new type
 						// serialization format.
@@ -137,6 +141,7 @@ func upgradeDescriptors(
 						}
 					}
 					key := catalogkeys.MakeDescMetadataKey(d.Codec, mut.GetID())
+					mut.MaybeIncrementVersion()
 					b.CPut(key, mut.DescriptorProto(), mut.GetRawBytesInStorage())
 				}
 				return txn.KV().Run(ctx, b)

--- a/pkg/upgrade/upgrades/first_upgrade_test.go
+++ b/pkg/upgrade/upgrades/first_upgrade_test.go
@@ -79,7 +79,8 @@ func TestFirstUpgrade(t *testing.T) {
 		"USE test",
 		"CREATE TABLE foo (i INT PRIMARY KEY, j INT, INDEX idx(j))",
 	)
-
+	// Select a timestamp for when the tables were created.
+	startTime := testServer.Clock().Now()
 	// Corrupt the table descriptor in an unrecoverable manner. We are not able to automatically repair this
 	// descriptor.
 	tbl := desctestutils.TestingGetPublicTableDescriptor(kvDB, keys.SystemSQLCodec, "test", "foo")
@@ -87,6 +88,7 @@ func TestFirstUpgrade(t *testing.T) {
 	require.NoError(t, kvDB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 		mut := tabledesc.NewBuilder(tbl.TableDesc()).BuildExistingMutableTable()
 		mut.NextIndexID = 1
+		mut.MaybeIncrementVersion()
 		return txn.Put(ctx, descKey, mut.DescriptorProto())
 	}))
 
@@ -106,6 +108,8 @@ func TestFirstUpgrade(t *testing.T) {
 	require.NoError(t, kvDB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 		mut := tabledesc.NewBuilder(tbl.TableDesc()).BuildExistingMutableTable()
 		mut.ModificationTime = hlc.Timestamp{}
+		mut.Indexes[0].NotVisible = true
+		mut.Version += 2 // Increment by 2 because we wrote a version above.
 		return txn.Put(ctx, descKey, mut.DescriptorProto())
 	}))
 
@@ -127,8 +131,9 @@ func TestFirstUpgrade(t *testing.T) {
 	// the MVCC timestamp.
 	require.False(t, readDescFromStorage().GetModificationTime().IsEmpty())
 	changes := readDescFromStorage().GetPostDeserializationChanges()
-	require.Equal(t, changes.Len(), 1)
+	require.Equal(t, 2, changes.Len())
 	require.True(t, changes.Contains(catalog.SetModTimeToMVCCTimestamp))
+	require.True(t, changes.Contains(catalog.SetIndexInvisibility))
 
 	// Wait long enough for precondition check to see the unbroken table descriptor.
 	execStmts(t, "CREATE DATABASE test3")
@@ -137,18 +142,18 @@ func TestFirstUpgrade(t *testing.T) {
 	// Upgrade the cluster version.
 	tdb.Exec(t, qUpgrade)
 
-	// The table descriptor protobuf should still have the modification time set;
-	// the only post-deserialization change should be SetModTimeToMVCCTimestamp.
+	// Run a schema change on the bar table.
+	tdb.Exec(t, "ALTER TABLE foo ADD COLUMN z INT")
+
+	// The only post-deserialization change should be SetModTimeToMVCCTimestamp,
+	// because the MVCC timestamp gets on each write.
 	require.False(t, readDescFromStorage().GetModificationTime().IsEmpty())
 	changes = readDescFromStorage().GetPostDeserializationChanges()
-	if v1.Equal(clusterversion.V25_4.Version()) {
-		// In 25.4, we do a one-time rewrite of all descriptors, so there should be
-		// no changes here. In later versions, there should be one change.
-		require.Equal(t, 0, changes.Len())
-	} else {
-		require.Equal(t, 1, changes.Len())
-		require.True(t, changes.Contains(catalog.SetModTimeToMVCCTimestamp))
-	}
+	require.Equal(t, 1, changes.Len())
+	require.True(t, changes.Contains(catalog.SetModTimeToMVCCTimestamp))
+	// Validate we can read bar before this upgrade. The lease manager should see
+	// distinct version numbers in the history. Otherwise, it can panic.
+	tdb.Exec(t, fmt.Sprintf("SELECT * FROM foo AS OF SYSTEM TIME '%s'", startTime.AsOfSystemTime()))
 }
 
 // TestFirstUpgradeRepair tests the correct repair behavior of upgrade
@@ -241,12 +246,14 @@ func TestFirstUpgradeRepair(t *testing.T) {
 			ConstraintID:        tbl.NextConstraintID,
 		}}
 		tbl.NextConstraintID++
+		tbl.MaybeIncrementVersion()
 		b.Put(catalogkeys.MakeDescMetadataKey(codec, tbl.GetID()), tbl.DescriptorProto())
 		fn := funcdesc.NewBuilder(fooFn.FuncDesc()).BuildExistingMutableFunction()
 		fn.DependedOnBy = []descpb.FunctionDescriptor_Reference{{
 			ID:        123456789,
 			ColumnIDs: []descpb.ColumnID{1},
 		}}
+		fn.MaybeIncrementVersion()
 		b.Put(catalogkeys.MakeDescMetadataKey(codec, fn.GetID()), fn.DescriptorProto())
 		return txn.Run(ctx, b)
 	}))
@@ -340,18 +347,34 @@ func TestFirstUpgradeRepair(t *testing.T) {
 	tdb.CheckQueryResults(t, qDetectRepairableCorruption, [][]string{{"0"}})
 	close(upgradeCompleted)
 	require.NoError(t, grp.Wait())
+	expectedVersionBump := 1
+	// Upgrades from before 25.4 will round-trip all descriptors once, so we
+	// expect two version bumps. Once for the repair and the next for the
+	// round trip.
+	if v0.Less(clusterversion.V25_4.Version()) {
+		expectedVersionBump += 1
+	}
 	// Assert that a version upgrade is reflected for repaired descriptors (stricly one version upgrade).
 	for _, d := range corruptDescs {
 		descId := d.GetID()
 		desc := readDescFromStorage(descId)
-		require.Equalf(t, descOldVersionMap[descId]+1, desc.GetVersion(), desc.GetName())
+		require.Equalf(t, descOldVersionMap[descId]+descpb.DescriptorVersion(expectedVersionBump), desc.GetVersion(), desc.GetName())
 	}
 
 	// Assert that no version upgrade is reflected for non-repaired descriptors.
+	expectedVersionBump = 0
+	// Upgrading from before 25.4 all non-database descriptors will be round tripped.
+	if v0.Less(clusterversion.V25_4.Version()) {
+		expectedVersionBump += 1
+	}
 	for _, d := range nonCorruptDescs {
 		descId := d.GetID()
 		desc := readDescFromStorage(descId)
-		require.Equalf(t, descOldVersionMap[descId], desc.GetVersion(), desc.GetName())
+		if desc.DescriptorType() != catalog.Database {
+			require.Equalf(t, descOldVersionMap[descId]+descpb.DescriptorVersion(expectedVersionBump), desc.GetVersion(), desc.GetName())
+		} else {
+			require.Equalf(t, descOldVersionMap[descId], desc.GetVersion(), desc.GetName())
+		}
 	}
 
 	// Check that the repaired table and function are OK.

